admin-manual/maintenance/common-atom-queries.rst

```diff
@@ -0,0 +1,2000 @@
+   fields related to other :term:`entity` types (such as authority records,
+   :term:`access points <access point>`, etc.) will not be stored directly in
+   these tables.
+
+   For a more detailed look at how AtoM's database is organized, see: 
+
+   * https://wiki.accesstomemory.org/Development/ERDs
+   * https://www.slideshare.net/accesstomemory/constructing-sql-queries-for-atom
+
+   See also: :ref:`sql-note-property-queries`
+
+If you'd like to see a count of all the :term:`archival description` records 
+currently in your database, you can use the following query: 
+
+.. code-block:: mysql
+
+   SELECT COUNT(*) FROM information_object_i18n AND id <>1;   
+
+We can also provide further criteria. For example, to count how many 
+descriptions include a Scope and content statement: 
+
+.. code-block:: mysql
+
+   SELECT COUNT(*) FROM information_object_i18n WHERE scope_and_content IS NOT NULL AND id <>1;
+
+Alternatively, we can search for all descriptions that are missing an Extent and 
+medium statement with: 
+
+.. code-block:: mysql
+
+   SELECT COUNT(*) FROM information_object_i18n WHERE extent_and_medium IS NULL AND id <>1;
+
+:ref:`Back to the top <common-atom-queries>`
+
+.. _sql-update-publication-status:
+
+Update draft archival descriptions to published
+-----------------------------------------------
+
+AtoM uses ID values in the status table to manage the :term:`publication status` 
+of archival descriptions. Typically, a status ID of 159 means draft, and 160 
+means published. Archival descriptions have a type_id value of 158. 
+
+We can therefore use the following query to publish all draft descriptions in AtoM:
+
+.. code:: bash
+
+   UPDATE status SET status_id=160 WHERE type_id=158 AND object_id <> 1;
+
+If we wanted to set all descriptions in AtoM to draft, we can change the ``status_id`` value
+of the above query from 160 to 159: 
+
+.. code:: bash
+
+   UPDATE status SET status_id=159 WHERE type_id=158 AND object_id <> 1;
+
+You will need to rebuild the search index after making publication status 
+changes. See: 
+
+* :ref:`maintenance-populate-search-index`
+
+**Updating all draft descriptions from a particular repository**
+
+We can also target just those descriptions that are associated with a particular
+:term:`repository` (AKA :term:`archival institution`). To do so, we first need 
+to retrieve the id of the repository, using the repository :term:`slug`. See 
+above for more information on how to get an object ID using a slug: 
+
+* :ref:`cli-object-id`
+
+Assuming in this example the id returned is 123, you can then run the following 
+query to perform the publication status updates, changing all descriptions 
+associated with this repository record to published:
+
+.. code:: bash
+
+   UPDATE status
+     SET status_id=160
+     WHERE type_id=158 AND status_id=159
+     AND object_id IN (
+       SELECT id FROM information_object
+       WHERE repository_id=123
+     );
```

Comment:
This won't update the descendants of those resources, as they inherit the repository through code logic.